phoenix/app/api_products/lib/api_products_web/services/products_service.ex

```diff
@@ -0,0 +1,39 @@
+defmodule ApiProductsWeb.ProductsService do
+  use ApiProductsWeb, :controller
+
+  alias ApiProducts.Management
+  alias ApiProductsWeb.RedisService
+  alias ApiProductsWeb.ElasticSearchService
+
+  def confirm_index() do
+    case RedisService.set_products(Management.list_products()) do
```

Comment:
Aqui não deveria retornar o valor presente no cache primeiro? Se não tiver presente aí sim, teria que buscar no banco e setar o cache.

Reply:
Com essa mudança tu vai ter que apagar a chave no cache sempre que algum produto for modificado (create/update/delete).